bin/ih-setup

```diff
@@ -301,20 +301,26 @@ function ih::private::state::is-step-installed() {
 }
 
 function ih::private::upgrade() {
-
   ih::log::info "Checking for updates..."
 
+  if ! brew update >/dev/null; then
+    ih::log::error "brew update failed"
+    return 1
+  fi
+
   local OUTDATED
-  brew update >/dev/null
   OUTDATED=$(brew outdated ih-core)
   if [ -z "$OUTDATED" ]; then
     ih::log::info "ih-setup is up-to-date"
     return 0
   fi
 
   ih::log::info "Upgrading ih-setup"
+  if ! brew upgrade ih-core; then
+    ih::log::error "brew upgrade ih-core failed"
+    return 1
```

Comment:
Seems like doing `return 1` instead of `exit 1` is fine since the code in `main` handles this.